src/client/sandbox/event/simulator.js

```diff
@@ -150,7 +150,7 @@ export default class EventSimulator {
             button:        options.button === void 0 ? eventUtils.BUTTON.left : options.button,
             buttons:       options.buttons === void 0 ? eventUtils.BUTTONS_PARAMETER.leftButton : options.buttons,
             relatedTarget: options.relatedTarget || null,
-            which:         options.which === void 0 ? eventUtils.WHICH_PARAMETER.leftButton : options.which
+            which:         options.which
```

Comment:
Did you check that we send `which` param for each simulated event? (for creation by `nativeMethods.documentCreateEvent`)

Reply:
We don't need to send it for each event, even created via `document.createEvent`. The `which` property is calculated correctly depending on `button` property. We need `which` param for events with not working `which` property, i.e. `mousemove`. So we pass `which` from testcafe to hammerhead on `mousemove`. But, I've mentioned that in any cases we do not have proper set of `button`, `buttons`, `which` prioperties for some events even now. Let's discuss it